internal/kubernetes/operator/dbusers/dbusers_test.go

```diff
@@ -0,0 +1,356 @@
+	t.Run("Can convert user labels from Atlas to the Operator format", func(t *testing.T) {
+		atlasUser := &mongodbatlas.DatabaseUser{
+			Labels: []mongodbatlas.Label{
+				{
+					Key:   "TestKey",
+					Value: "TestValue",
+				},
+			},
+		}
+
+		expectedLabels := []common.LabelSpec{
+			{
+				Key:   "TestKey",
+				Value: "TestValue",
+			},
+		}
+
+		if got := convertUserLabels(atlasUser); !reflect.DeepEqual(got, expectedLabels) {
+			t.Errorf("convertUserLabels() = %v, want %v", got, expectedLabels)
+		}
+	})
+}
+
+func Test_convertUserRoles(t *testing.T) {
+	t.Run("Can convert user labels from Atlas to the Operator format", func(t *testing.T) {
+		atlasUser := &mongodbatlas.DatabaseUser{
+			Roles: []mongodbatlas.Role{
+				{
+					RoleName:       "TestRole",
+					DatabaseName:   "TestDB",
+					CollectionName: "TestCollection",
+				},
+			},
+		}
+
+		expectedRoles := []atlasV1.RoleSpec{
+			{
+				RoleName:       "TestRole",
+				DatabaseName:   "TestDB",
+				CollectionName: "TestCollection",
+			},
+		}
+		if got := convertUserRoles(atlasUser); !reflect.DeepEqual(got, expectedRoles) {
+			t.Errorf("convertUserRoles() = %v, want %v", got, expectedRoles)
+		}
+	})
+}
+
+func Test_buildUserSecret(t *testing.T) {
+	t.Run("Can build user secret WITHOUT credentials", func(t *testing.T) {
+		projectName := "TestProject-1"
+		atlasUser := &mongodbatlas.DatabaseUser{
+			Password: "TestPassword",
+			Username: "TestName",
+		}
+
+		expectedSecret := &corev1.Secret{
+			TypeMeta: v1.TypeMeta{
+				Kind:       "Secret",
+				APIVersion: "v1",
+			},
+			ObjectMeta: v1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-%s", projectName, atlasUser.Username),
+				Namespace: "TestNamespace",
+				Labels: map[string]string{
+					secrets.TypeLabelKey: secrets.CredLabelVal,
+				},
+			},
+			Data: map[string][]byte{
+				PasswordField: []byte(""),
+			},
+		}
+		if got := buildUserSecret(projectName, atlasUser, "TestNamespace", false); !reflect.DeepEqual(got, expectedSecret) {
+			t.Errorf("buildUserSecret(); \r\n got:%v;s\r\n want:%v", got, expectedSecret)
+		}
+	})
+
+	t.Run("Can build user secret WITH credentials", func(t *testing.T) {
+		projectName := "TestProject-2"
+		atlasUser := &mongodbatlas.DatabaseUser{
+			Password: "TestPassword",
+			Username: "TestName",
+		}
+
+		expectedSecret := &corev1.Secret{
+			TypeMeta: v1.TypeMeta{
+				Kind:       "Secret",
+				APIVersion: "v1",
+			},
+			ObjectMeta: v1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-%s", projectName, atlasUser.Username),
+				Namespace: "TestNamespace",
+				Labels: map[string]string{
+					secrets.TypeLabelKey: secrets.CredLabelVal,
+				},
+			},
+			Data: map[string][]byte{
+				PasswordField: []byte(atlasUser.Password),
+			},
+		}
+		if got := buildUserSecret(projectName, atlasUser, "TestNamespace", true); !reflect.DeepEqual(got, expectedSecret) {
+			t.Errorf("buildUserSecret(); \r\n got:%v;s\r\n want:%v", got, expectedSecret)
+		}
+	})
```

Comment:
[q] can this be written as a test case matrix? example
https://github.com/mongodb/mongodb-atlas-cli/blob/3d0bea5f6b288cc3b4dc7447748f7a0fa8c59c21/internal/cli/atlas/backup/schedule/update_test.go#L77-L114

Reply:
It was like that for all tests. But I decided to follow other unit tests style.